From 127ac4c8de64e954dbb4d1cf9b5e365639796cc7 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Thu, 22 Feb 2024 08:48:55 +0100
Subject: [PATCH 1/1] tmpfiles: add and install for uuidd, generate /run/uuidd
 & /var/lib/libuuid

These directories are requird for uuidd, so let systemd-tmpfiles create them.

Signed-off-by: Christian Hesse <mail@eworm.de>
(cherry picked from commit 2b7410544c90f56fbb9abca999ed48feffbe31ef)
---
 misc-utils/.gitignore             | 1 +
 misc-utils/Makemodule.am          | 4 +++-
 misc-utils/meson.build            | 8 ++++++++
 misc-utils/uuidd-tmpfiles.conf.in | 6 ++++++
 4 files changed, 18 insertions(+), 1 deletion(-)
 create mode 100644 misc-utils/uuidd-tmpfiles.conf.in

diff --git a/misc-utils/.gitignore b/misc-utils/.gitignore
index c0287917d..6ceec2057 100644
--- a/misc-utils/.gitignore
+++ b/misc-utils/.gitignore
@@ -3,3 +3,4 @@ uuidd.8
 uuidd.rc
 uuidd.service
 uuidd.socket
+uuidd-tmpfiles.conf
diff --git a/misc-utils/Makemodule.am b/misc-utils/Makemodule.am
index 71548c9f9..05804ccff 100644
--- a/misc-utils/Makemodule.am
+++ b/misc-utils/Makemodule.am
@@ -128,6 +128,7 @@ uuidd_SOURCES = misc-utils/uuidd.c lib/monotonic.c lib/timer.c
 if HAVE_SYSTEMD
 uuidd_LDADD += $(SYSTEMD_LIBS) $(SYSTEMD_DAEMON_LIBS)
 uuidd_CFLAGS += $(SYSTEMD_CFLAGS) $(SYSTEMD_DAEMON_CFLAGS)
+tmpfiles_DATA += misc-utils/uuidd-tmpfiles.conf
 systemdsystemunit_DATA += \
 	misc-utils/uuidd.service \
 	misc-utils/uuidd.socket
@@ -142,7 +143,8 @@ endif # BUILD_UUIDD
 PATHFILES += \
 	misc-utils/uuidd.rc \
 	misc-utils/uuidd.service \
-	misc-utils/uuidd.socket
+	misc-utils/uuidd.socket \
+	misc-utils/uuidd-tmpfiles.conf
 
 if BUILD_BLKID
 sbin_PROGRAMS += blkid
diff --git a/misc-utils/meson.build b/misc-utils/meson.build
index 7d21d02c1..026d92d11 100644
--- a/misc-utils/meson.build
+++ b/misc-utils/meson.build
@@ -76,6 +76,14 @@ test_uuidd_sources = files(
 )
 
 if build_uuidd and systemd.found()
+  uuidd_tmpfiles = configure_file(
+    input : 'uuidd-tmpfiles.conf.in',
+    output : 'uuidd-tmpfiles.conf',
+    configuration : conf)
+  install_data(
+    uuidd_tmpfiles,
+    install_dir : '/usr/lib/tmpfiles.d')
+
   uuidd_service = configure_file(
     input : 'uuidd.service.in',
     output : 'uuidd.service',
diff --git a/misc-utils/uuidd-tmpfiles.conf.in b/misc-utils/uuidd-tmpfiles.conf.in
new file mode 100644
index 000000000..9c2005bdf
--- /dev/null
+++ b/misc-utils/uuidd-tmpfiles.conf.in
@@ -0,0 +1,6 @@
+# This file is part of uuidd.
+#
+# See tmpfiles.d(5) for details
+#
+d @runstatedir@/uuidd 2775 uuidd uuidd -
+d /var/lib/libuuid 0755 uuidd uuidd -
-- 
2.44.0

